src/main/java/uk/nhs/digital/nhsconnect/nhais/service/edifact_to_fhir/DeductionTransactionMapper.java

```diff
@@ -2,14 +2,51 @@
 
 import org.hl7.fhir.r4.model.Parameters;
 import org.springframework.stereotype.Component;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.DeductionDate;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.DeductionReasonCode;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.NewHealthAuthorityName;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.PersonName;
 import uk.nhs.digital.nhsconnect.nhais.model.edifact.ReferenceTransactionType;
 import uk.nhs.digital.nhsconnect.nhais.model.edifact.Transaction;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.message.EdifactValidationException;
+import uk.nhs.digital.nhsconnect.nhais.model.fhir.NhsIdentifier;
+import uk.nhs.digital.nhsconnect.nhais.model.fhir.ParameterNames;
+import uk.nhs.digital.nhsconnect.nhais.model.fhir.ParametersExtension;
+
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Optional;
 
 @Component
 public class DeductionTransactionMapper implements FhirTransactionMapper {
+
+    private final static DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+
     @Override
     public void map(Parameters parameters, Transaction transaction) {
-        //TODO: to be implemented
+        transaction.getPersonName()
+            .map(PersonName::getNhsNumber)
+            .flatMap(this::mapToNhsIdentifier)
```

Comment:
If NHS number is null, it will result with empty optional so why wrap it into an `Optional.ofNullable' here https://github.com/nhsconnect/integration-adaptor-nhais/pull/168/files#diff-3f885969ed5b91a86b2341f25af3020cR49 and flatmap it here?
This can be 
```
transaction.getPersonName()
    .map(PersonName::getNhsNumber)
    .filter(StringUtils::isNotBlank)
    .map(NhsIdentifier::new)
```
and IMO would be more readable

Reply:
fixed